### PR TITLE
Fix ordering of the typefaces list 

### PR DESCRIPTION
**Pull request by shaoshing**
This PR addresses the following issues:

1 typeface should ordered case insensitively. (#2530)

screen shot 2015-10-18 at 10 47 39 am

2 some typefaces are not ordered by title. This is becuase we ordered typeface by postScriptName, and it turns out that `title` does not always match with `postScriptName`. For example `title: Wide Latin, postScriptName: LatinWide`, `title: Wawati SC Regular, postScriptName: DFWaWaSC-W5`.

screen shot 2015-10-18 at 11 24 45 am

The fix is to sort typefaces by upper case title, instead of postScriptName. However, it breaks the binary search in `Select._findIndex`, as postScriptNames (which are the ids of of Select's options) are longer ordered. So another change I made is replacing the ID with typeface index (zero padding string). I have several solutions, and this is the one that requires least changes. Do let me know if you have better solution :D

### CODE REVIEW

File: src/js/jsx/sections/style/Type.jsx [hunk 1]
```diff
@@ -142,12 +142,16 @@ define(function (require, exports, module) {
...
-         * Set the type face of the selected text layers from a font's postscript name.
+         * Set the type face of the selected text layers from a font's ID.
          * 
          * @private
-         * @param {string} postScriptName
+         * @param {string} typefaceID
          */
-        _handleTypefaceChange: function (postScriptName) {
+        _handleTypefaceChange: function (typefaceID) {
+            var typefaceIndex = parseInt(typefaceID), // typefaceID is equivalent to index in this.state.typefaces
```
Comment (iwehrman) [inline src/js/jsx/sections/style/Type.jsx]: Drive-by nit: we should never use `window.parseInt`. It's harmless here, but in most other cases the fact that it returns a `NaN` when the number doesn't parse is a real bummer. (We could probably talk JSHint or JSCS into flagging this as an error.) Try `parseNumber` from `js/util/math` instead.

**Comment by shaoshing** [comment]
@iwehrman thanks for reviewing. I've replace `parseInt` with `math.parseNumber`. Now back to you.

**Comment by iwehrman** [comment]
In the future, we should just provide a comparator to `Datalist` for this kind of thing. But, for now, this'll do.
